play-json/shared/src/main/scala/play/api/libs/json/JsonConfiguration.scala

```diff
@@ -11,15 +11,21 @@ sealed trait JsonConfiguration {
 
   /** Naming strategy */
   def naming: JsonNaming
+
+  /** How options are handled by the macro */
+  def optionHandlers: OptionHandlers = OptionHandlers.Default
 }
 
 object JsonConfiguration {
   type Aux[O <: Json.MacroOptions] = JsonConfiguration { type Opts = O }
 
   private final class Impl[O <: Json.MacroOptions](
-    val naming: JsonNaming = JsonNaming.Identity
+    val naming: JsonNaming = JsonNaming.Identity,
+    override val optionHandlers: OptionHandlers = OptionHandlers.Default
```

Comment:
Would not put `override` modifier there

Reply:
The modifier is necessary here:
`overriding method optionHandlers in trait JsonConfiguration of type => play.api.libs.json.OptionHandlers`

I wanted to not set `optionHandlers` in the trait but then it failed the binary compatibility check.

Reply:
As the implementation is provided in the trait as `OptionHandlers.Default`, and as this `Impl` class is not part of the public API (as `private`), I would add a filter in MiMa (would found it cleaner than `override`).

Reply:
Binary compatibility is hard, I'll take an other look at it later

Reply:
You can copy the warning when there is no such `override` and paste it in the MiMa filters already in `build.sbt` for `play-json` module

Reply:
Would also add `ProblemFilters.exclude[ReversedMissingMethodProblem]("play.api.libs.json.JsonConfiguration.optionHandlers")` as having a new function is not breaking the API compat for the existing projects using Play JSON.